src/snowflake/cli/_app/telemetry.py

```diff
@@ -81,6 +86,33 @@ class TelemetryEvent(Enum):
 TelemetryDict = Dict[Union[CLITelemetryField, TelemetryField], Any]
 
 
+def _is_cli_exception(exception: Exception) -> bool:
+    return issubclass(
+        exception.__class__,
+        (click.ClickException, typer.Exit, typer.Abort, BrokenPipeError),
+    )
+
+
+def _get_additional_exception_information(exception: Exception) -> TelemetryDict:
+    """
+    Attach the errno and sqlstate if the exception or the
+    cause of the exception is a ProgrammingError
+    """
+    additional_info = {}
+
+    if isinstance(exception, ProgrammingError):
+        additional_info[CLITelemetryField.ERROR_CODE] = exception.errno
+        additional_info[CLITelemetryField.SQL_STATE] = exception.sqlstate
+
+    if isinstance(exception.__cause__, ProgrammingError):
+        cause = exception.__cause__
+        additional_info[CLITelemetryField.ERROR_CAUSE] = cause.__class__.__name__
```

Comment:
how come we don't add this in the first `if`?

Reply:
Marcus, see previous comment for the suggested flow.